samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -249,6 +251,44 @@ public void waitForFinish() {
     }
   }
 
+
+  private void createIntermediateStreams(List<StreamSpec> intStreams) throws Exception {
+    boolean streamsExist = getStreamManager().checkIfStreamsExist(intStreams);
+    if (!streamsExist) {
+      getStreamManager().createStreams(intStreams);
+    }
+  }
+
+  /**
+   * Create intermediate streams using {@link org.apache.samza.execution.StreamManager}.
+   * If {@link CoordinationUtils} is provided, this function will first acquire a lock, and then create the streams.
+   * All the runner processes will wait till the time they acquire the lock. On acquiring lock, they will check if the streams have already been created.
+   * If streams exist already, they will unlock and proceed normally. If streams don't exist, they create the streams and then unlock.
+   * @param intStreams list of intermediate {@link StreamSpec}s
+   * @throws Exception exception for latch timeout
+   */
+  private void createStreamsWithLock(List<StreamSpec> intStreams) throws Exception {
+    if (!intStreams.isEmpty()) {
```

Comment:
nit. I prefer if(intStreams.isEmpty()) return;  to avoid unnecessary indentation. But that's personal preference.